### PR TITLE
Cmyk pdf feature

### PR DESCRIPTION
**Pull request by xavierraffin**
CMYK pdf are needed by printers like offset or web (rotatives)

### CODE REVIEW

**Comment by datanel** [comment]
your PR depends on GhostScript
Can you update the REAME please?

File: web/index.php [hunk 1]
```diff
@@ -27,12 +27,40 @@
...
     header('Content-Type: application/pdf');
-    echo $snappy->getOutput($url);
+
+    if (isset($_GET['cmyk'])) {
+        // Convert pdf in CMYK colorspace
+        // Need GhostScript
+        // Need a writeable /tmp directory
+        $rand_id = rand();
+        $tmpRGBFileName = '/tmp/temppdf-rgb-'.$rand_id.'.pdf';
+        $tmpCMYKFileName = '/tmp/temppdf-cmyk-'.$rand_id.'.pdf';
```
Comment (vinzentt) [inline web/index.php]: maybe to find the writable temp dir and be more OS compatible, you should rely on http://php.net/manual/fr/function.sys-get-temp-dir.php

**Comment by xavierraffin** [comment]
It's done

File: README.md [hunk 2]
```diff
@@ -32,16 +35,42 @@ knplabs/knp-snappy: https://github.com/KnpLabs/snappy
...
+
 All these parameters may be passed to the webservice:
-http://localhost/pdfGenerator/web/index.php?url=google.fr&orientation=landscape&zoom=2
+
+``http://localhost/pdfGenerator/web/index.php?url=google.fr&orientation=landscape&zoom=2``
+
+Options
+-------------
+
+Optionnal url get parameters could be send to pdfGenerator:
```
Comment (datanel) [inline README.md]: sent

File: README.md [hunk 3]
```diff
@@ -32,16 +35,42 @@ knplabs/knp-snappy: https://github.com/KnpLabs/snappy
...
+
+* __margin__ : Will set CSS document sheet margin
+* __cmyk__ : When set to 1 or True, the returned pdf will use CMYK colorspace (usefull for offset or Web proffessionnal printers)
```
Comment (datanel) [inline README.md]: professional ? not sure

File: web/index.php [hunk 4]
```diff
@@ -27,12 +27,39 @@
...
-    
+
     // Display the resulting pdf in the browser
     // by setting the Content-type header to pdf
     // header('Content-Disposition: attachment; filename="file.pdf"');
     header('Content-Type: application/pdf');
-    echo $snappy->getOutput($url);
+
+    if (isset($_GET['cmyk'])) {
+        // Convert pdf in CMYK colorspace
```
Comment (datanel) [inline web/index.php]: you should write this comment before the if

**Comment by xavierraffin** [comment]
Done

File: web/index.php [hunk 5]
```diff
@@ -27,12 +27,40 @@
...
     // Display the resulting pdf in the browser
     // by setting the Content-type header to pdf
     // header('Content-Disposition: attachment; filename="file.pdf"');
     header('Content-Type: application/pdf');
-    echo $snappy->getOutput($url);
+
+    // Convert pdf in CMYK colorspace
+    // Need GhostScript
+    // Need a writeable temporary directory for php process
+    if (isset($_GET['cmyk'])) {
```
Comment (datanel) [inline web/index.php]: you say in the Readme "When set to 1 or True" but in this condition you can set any value
you should add 
```php
if (isset($_GET['cmyk']) && $_GET['cmyk'] === 1) {
   ...
```

**Comment by datanel** [comment]
I have tested and it is well converted

File: web/index.php (hunk 5 above)
Comment (xavierraffin) [inline web/index.php]: lol.  
Note that my README instruction works.
But OK if you don't sleep because of this I just change it for you.

Comment (datanel) [inline web/index.php]: Your instruction works perfectly but it's just to be consistent with what you say in the README.
If you say cmyk must be 1 or true and just let isset($_GET['cmyk']), it works but it also works with any value, not just 1 or true. 

It's not just for me

**Pull request merged**